### PR TITLE
[breaking] Implement the es-shim-api interface

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
1. I changed the package so that it follows the `es-shim-api` interface. Since it requires multiple files, it's now harder to directly use this package in the browser. However, there is an easy (even if not recommended in production) workaround using `bundle.run`.
2. Instead of inlining the abstract operations and the `defineProperty` IE check, I used the `es-abstract` and `define-properties` packages. This has the disadvantage of a new dependency, but they have 20M and 13M dowloads/month on npm so they are likely already in the `node_modules` of our users.
3. I changed the tests to provide a complete output in case of multiple failures, and added every node version to travis.

cc @ljharb

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
The Travis build that's marked as pending passed: https://travis-ci.org/github/mathiasbynens/String.prototype.includes/builds/688585269

File: index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+'use strict';
+
+var callBind = require('es-abstract/helpers/callBind');
+var define = require('define-properties');
+
+var implementation = require('./implementation');
+var getPolyfill = require('./polyfill');
+var shim = require('./shim');
+
+var boundIncludes = callBind(implementation);
```
Comment (ljharb) [inline index.js:12]:
```suggestion
var boundIncludes = callBind(getPolyfill());
```

File: tests/tests.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ var assert = require('assert');
 var assertEquals = assert.equal;
 var assertThrows = assert['throws'];
 
-require('../includes.js');
+require('../auto.js');
```
Comment (ljharb) [inline tests/tests.js]:
```suggestion
require('../auto');
```
also since that's what's in "exports"

File: package.json [hunk 3]
```diff
@@ -21,12 +29,21 @@
...
 	},
 	"bugs": "https://github.com/mathiasbynens/String.prototype.includes/issues",
-	"files": [
-		"LICENSE-MIT.txt",
-		"includes.js"
-	],
 	"scripts": {
-		"test": "node tests/tests.js",
-		"cover": "istanbul cover --report html --verbose --dir coverage tests/tests.js"
+		"pretest": "es-shim-api --bound",
```
Comment (nicolo-ribaudo) [inline package.json:33]: I should remove `--bound`, right?

Comment (ljharb) [reply package.json:33]: no, at the moment, it requires it because the index isn't `===` to the polyfill.

**Pull request merged**